akvo/rsr/spa/app/modules/editor/project-init-handler.js

```diff
@@ -64,26 +66,30 @@ const ProjectInitHandler = connect(({editorRdr}) => ({ editorRdr }), actions)(Re
     }
     else resolve()
   })
-  let nextSectionIndex = 0
-  const fetchNextSection = () => {
-    fetchSection(sectionIndexPipeline[nextSectionIndex])
-    .then(() => {
-      props.setSectionFetched(sectionIndexPipeline[nextSectionIndex])
-      if (nextSectionIndex < 10) {
-        nextSectionIndex += 1
-        fetchNextSection()
-      }
-    })
-  }
+
   useEffect(() => {
-    if (prevParams && prevParams.id !== params.id && params.id !== 'new'){
-      fetchSection(3)
-      fetchSection(5)
+    // eslint-disable-next-line no-restricted-globals
+    if (!isNaN(params?.id) && (nextSectionIndex < 11)) {
+      if (!props?.editorRdr[`section${nextSectionIndex + 1}`]?.isFetched) {
+        fetchSection(sectionIndexPipeline[nextSectionIndex])
+          .then(() => {
+            if (nextSectionIndex === 4) {
+              /**
+               * Results and indicator need 10s delay to avoid fields component errors.
+               */
+              setTimeout(() => {
+                props.setSectionFetched(sectionIndexPipeline[nextSectionIndex])
+              }, 10000)
```

Comment:
Why are there field component errors? If it's too long for this PR, then a new issue should be created to treat this later and the issue number mentioned in this comment

Reply:
Because some field arrays are rendered first before the data state is ready and I can't find an area to indicate the state is ready. So, that's why I delayed to set fetched status as true when we got success response from backend.